### PR TITLE
EZP-28844: Fix cluster support in docker and Platform.sh

### PR DESCRIPTION
**Pull request by vidarl**
This PR makes it convenient to config cluster settings when using docker or platform.sh
by cluster config I mean:
- session stored in redis
- cache stored in redis
- enabling dfs file handler

### CODE REVIEW

File: app/config/env/platformsh.php [hunk 1]
```diff
@@ -18,12 +18,44 @@
     $container->setParameter('database_name', $endpoint['path']);
     $container->setParameter('database_user', $endpoint['username']);
     $container->setParameter('database_password', $endpoint['password']);
-    $container->setParameter('database_path', '');
 
+    // 'cluster_database_name' is deprecated in eZ Platform 2.1
```
Comment (vidarl) [inline app/config/env/platformsh.php]: Yeah, I want to deprecate this one. I guess 2.1 is the first upcoming version where we may do so?

File: app/config/env/platformsh.php [hunk 2]
```diff
@@ -18,21 +21,56 @@
...
         if ($endpoint['scheme'] !== 'redis') {
             continue;
         }
 
+        $container->setParameter('cache_pool', 'singleredis');
         $container->setParameter('cache_host', $endpoint['host']);
         $container->setParameter('cache_redis_port', $endpoint['port']);
+
+        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__ . '/../cache_pool'));
+        $loader->load('singleredis.yml');
```
Comment (andrerom) [inline app/config/env/platformsh.php]: prior attempt was to let generic.php take care of the loading by specifying the needed env variables.

File: app/config/env/platformsh.php (hunk 1 above)
Comment (andrerom) [reply app/config/env/platformsh.php]: Should be fine, can also write 1.13.1/2.1

Comment (vidarl) [reply app/config/env/platformsh.php]: ok. May I add, ", removed in 2.2" ?

File: app/config/env/platformsh.php (hunk 2 above)
Comment (vidarl) [reply app/config/env/platformsh.php]: well.
In generic.php it loads the yml if `$CUSTOM_CACHE_POOL` is set, which is fine.

So, we could require platform.sh users to also set that environment variable in order to load the yaml, but the rest of the logic will anyway be different as `cache_host` does not have a fixed value on platform.sh. It needs to be fetched from `$PLATFORM_RELATIONSHIPS` anyway.

Thus, I think it is better that we also autodetect the whole thing on platform.sh using `$PLATFORM_RELATIONSHIPS`

File: app/config/env/platformsh.php (hunk 1 above)
Comment (andrerom) [reply app/config/env/platformsh.php]: Then you can just as well remove it in master (2.1)

**Comment by vidarl** [comment]
> Besides inline comment, do we also need change on apache vhost for the dfs change?

apache template fixed

**Comment by andrerom** [comment]
Looks good, should there perhaps be a JIRA issue number mentioned in the PR title and linked to in description?

File: app/config/env/platformsh.php (hunk 2 above)
Comment (andrerom) [reply app/config/env/platformsh.php]: ok, then make sure we are consistent on this.

Comment (vidarl) [reply app/config/env/platformsh.php]: so, you approve then?

We could keep the load() code only in generic.php by changing the condition for it's execution. Something like `if $loadSingleRedisYml then ....`, and then set `$loadSingleRedisYml = true;` when needed in both generic.php and platformsh.php

Comment (andrerom) [reply app/config/env/platformsh.php]: yes, it's ok as is, we can cleanup if needed once the other things are in.

**Comment by vidarl** [comment]
Squashed commit and added jira issue

**Pull request merged**